container/docker/factory.go

```diff
@@ -49,6 +49,8 @@ var dockerCgroupRegexp = regexp.MustCompile(`.+-([a-z0-9]{64})\.scope$`)
 
 var noSystemd = flag.Bool("nosystemd", false, "Explicitly disable systemd support for Docker containers")
 
+var dockerMetadataEnvs = flag.String("docker_metadata_env", "", "Comma seperated list with names of env variables, which will be exported as metadata (default: empty)")
```

Comment:
How about `--docker-env-metadata-whitelist: a comma-separated list of environment variable keys that needs to be collected for docker containers`?
And BTW will this default to all env variables? To me that feels like a good default.